test/gtest/common/mem_buffer.cc

```diff
@@ -145,6 +149,37 @@ void mem_buffer::set_device_context()
     device_set = true;
 }
 
+size_t mem_buffer::get_bar1_free_size()
+{
+    /* All gtest CUDA tests explicitly assume that all memory allocations are
+     * done on the device 0. The same assumption is followed here. */
+    size_t max_availible_size = SIZE_MAX;
+
+#if HAVE_NVML_H
+    nvmlReturn_t ret;
+    nvmlDevice_t device;
+    nvmlBAR1Memory_t bar1mem;
+
+    ret = nvmlDeviceGetHandleByIndex(0, &device);
+    if (ret != NVML_SUCCESS) {
+        /* For whatever reason we cannot open device handle.
+         * As a result let's assume there is no limit on the size
+         * and in the worse case scenario gtest will fail in runtime */
+        return max_availible_size;
+    }
+
+    ret = nvmlDeviceGetBAR1MemoryInfo(device, &bar1mem);
+    if (ret != NVML_SUCCESS) {
+        /* Similarly let's assume there is no limit on the size */
+        return max_availible_size;
+    }
+
+    max_availible_size = (size_t)bar1mem.bar1Free;
```

Comment:
IMO because of this line need to rename max_availible_size to availible_size
also, can we cache this value in a static (global) variable?

Reply:
I have not seen failures on my machine with small bar1 but I do have to some other failures around asymetic connectivity.